src/nvme/ioctl.h

```diff
@@ -335,6 +335,59 @@ enum nvme_cmd_dword_fields {
 	NVME_DSM_CDW11_IDW_MASK					= 0x1,
 	NVME_DSM_CDW11_AD_SHIFT					= 2,
 	NVME_DSM_CDW11_AD_MASK					= 0x1,
+	NVME_DSM_CDW11_ATTRS_SHIFT				= NVME_DSM_CDW11_IDR_SHIFT,
+	NVME_DSM_CDW11_ATTRS_MASK				= NVME_VAL(DSM_CDW11_IDR) |
+								  NVME_VAL(DSM_CDW11_IDW) |
+								  NVME_VAL(DSM_CDW11_AD),
+	NVME_NVM_CDW2_ELBTU_SHIFT				= 0,
+	NVME_NVM_CDW2_ELBTU_MASK				= 0xffff,
+	NVME_NVM_CDW3_ELBTU_SHIFT				= 0,
+	NVME_NVM_CDW3_ELBTU_MASK				= 0xffffffff,
+	NVME_NVM_CDW10_SLBAL_SHIFT				= 0,
+	NVME_NVM_CDW10_SLBAL_MASK				= 0xffffffff,
+	NVME_NVM_CDW11_SLBAU_SHIFT				= 0,
+	NVME_NVM_CDW11_SLBAU_MASK				= 0xffffffff,
+	NVME_NVM_CDW12_NLB_SHIFT				= 0,
+	NVME_NVM_CDW12_NLB_MASK					= 0xffff,
+	NVME_NVM_CDW12_CETYPE_SHIFT				= 16,
+	NVME_NVM_CDW12_CETYPE_MASK				= 0xf,
+	NVME_NVM_CDW12_STC_SHIFT				= 24,
+	NVME_NVM_CDW12_STC_MASK					= 0x1,
+	NVME_NVM_CDW12_PRINFO_SHIFT				= 26,
+	NVME_NVM_CDW12_PRINFO_MASK				= 0xf,
+	NVME_NVM_CDW12_FUA_SHIFT				= 30,
+	NVME_NVM_CDW12_FUA_MASK					= 0x1,
+	NVME_NVM_CDW12_LR_SHIFT					= 31,
+	NVME_NVM_CDW12_LR_MASK					= 0x1,
+	NVME_NVM_CDW12_CONTROL_SHIFT				= NVME_NVM_CDW12_CETYPE_SHIFT,
+	NVME_NVM_CDW12_CONTROL_MASK				= NVME_VAL(NVM_CDW12_CETYPE) |
+								  NVME_VAL(NVM_CDW12_STC) |
+								  NVME_VAL(NVM_CDW12_PRINFO) |
+								  NVME_VAL(NVM_CDW12_FUA) |
+								  NVME_VAL(NVM_CDW12_LR),
```

Comment:
btw, this masks is already shifted thus it doesn't work as expected. I've added the necessary `>> NVME_NVM_CDW12_CONTROL_SHIFT` so it works. Caught by `meson test -C .build` :)